Homework #3 - Pull Request

lee-112 opened the pull request:
# Homework #3 - Pull Request

This pull request includes the code solution for the homework exercises written in HTML, CSS, JavaScript, and React.
Here are the improvements made to the pull request:

## Hw3/01:
1. Used Bootstrap & CSS styling for alignment, color, text, and sizing.
2. Used `try/catch` for error handling. 
3. Used `fetch` for API call  `Async/Await` to handle promises.
4. Template literals to display JSON data.

## Hw3/02:
1.  Used Bootstrap & CSS styling for alignment, color, text, and sizing.
2. Used `try/catch` for error handling.
3. Used `fetch` for API call  `Async/Await` to handle promises.
4. Defined `Object` to store doughnut chart labels and data.
6. Implemented function to filter characters by house name using js:
    1. Methods `forEach`, `includes`, and `switch`

## Hw3/03-React: ([codesandbox.io](www.codesandbox.io))
1. Used Bootstrap & CSS styling for alignment, color, text, and sizing.
2. Folder `/src/App.jsx`  Added import files and navbar paths.
3. Index.js Added bootstrap and react imports and elements needed to initialize and render the React application.
4. created folder `/src/components` for all React components.
	1. Files: `Navbar.jsx`, `Search.jsx`, `Houses.jsx`, and `Home`

5. `Home`: contains a Home function that returns home page information.
6. `Navbar.jsx`: Contains a Navbar function that returns element tags for the navigation bar.
7. `Search`: Contains a Search function that returns element tags for input button and character information.
8. `Houses`: Contains Houses function that returns a doughnut chart with all of the character houses.
<br>

gsampang commented on hw3/01-characters.js:
```diff
@@ -1,4 +1,27 @@
+  data.forEach((character) => {
+    const characterCards = `
+      <div class="card">
+        <img src="${character.imageUrl}">
```
I ran your code on my system and provided some feedback below. Overall, it looks good. 
I have focused my feedback on part 1 and 2 since these are the parts I understand the most.

Regarding 1:
- Running a WAVE check on it reveals 53 errors, all of which could easily be fixed by adding alt text to the images.
- Looking at the code, why are the name and title in heading 5 and heading 6 respectively? Headings 2-4 appear to have been skipped, and this comes up as an alert in WAVE.

Regarding 2:
- No changes necessary, just noticed the difference in granularity between yours and mine. Interesting the way you grouped many families together as other.

anhoop89 commented:
**01-Character:**
- Gabriel mentioned your Wave issues: 
- Everything else looks good to me. 
![image](https://github.com/lee-112/frontend-homework/assets/102849461/d11ee197-b222-4f47-bcfe-bace8e21a2fb)
 **02-Chart** 
- This one is perfect! Ready to merge :D 
**03-React**
- Home: there 2 WAVE warnings for this route.  A redundant link means you have logo and home linked to /home. 
![image](https://github.com/lee-112/frontend-homework/assets/102849461/b96241f9-e235-4b4b-8900-1ebaf1af9ebf)
- Search: Worked properly. Minor issues with WAVE
![image](https://github.com/lee-112/frontend-homework/assets/102849461/a47de6ff-19f3-42a6-b77d-92f74aa4fb2c)
- Houses:  This one is perfect. Minor issues from WAVE because it catches from the navbar. 

Great Job and clean code as always ( :

lee-112 commented:
# Homework #3 - Updated Pull Request

I have updated the code from the PR feedback.

## hw3/01-Characters.js

### Update-1: 
Fixed header tag ordering by h5 and h6 to h2 and h3 tags.
Updated the h2 and h3 class tags w/ bootstrap front styling.

### Update-2:
Fixed image alternative text error by adding alt attribute.
Updated alt attribute text w/ character id number.

anhoop89 commented:
**03-React: Search.jsx** 
![image](https://github.com/lee-112/frontend-homework/assets/102849461/287fbbe4-00d1-4c33-a7df-53f243b446c6)
It worked properly 
However, there are a couple of existing Wave issues. 
![image](https://github.com/lee-112/frontend-homework/assets/102849461/2a5d5b39-92ec-412c-bd38-3f913daa3964)
![image](https://github.com/lee-112/frontend-homework/assets/102849461/5e32f723-5314-43bd-a21d-e8d0b1283012)
